webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PolicyWriterService.scala

```diff
@@ -264,20 +262,29 @@ class PolicyWriterServiceImpl(
     , generationTime  : DateTime
     , parallelism     : Int
   ) : Box[Seq[NodeConfiguration]] = {
+    writeTemplatePure(rootNodeId, nodesToWrite, allNodeConfigs, versions, globalPolicyMode, generationTime, parallelism).toBox
+  }
+
+  def writeTemplatePure(
+      rootNodeId      : NodeId
+    , nodesToWrite    : Set[NodeId]
+    , allNodeConfigs  : Map[NodeId, NodeConfiguration]
+    , versions        : Map[NodeId, NodeConfigId]
+    , globalPolicyMode: GlobalPolicyMode
+    , generationTime  : DateTime
+    , parallelism     : Int
+  ) : IOResult[Seq[NodeConfiguration]] = {
 
     val nodeConfigsToWrite     = allNodeConfigs.filterKeys(nodesToWrite.contains(_))
     val interestingNodeConfigs = allNodeConfigs.filterKeys(k => nodeConfigsToWrite.exists{ case(x, _) => x == k }).values.toSeq
     val techniqueIds           = interestingNodeConfigs.flatMap( _.getTechniqueIds ).toSet
 
     //debug - but don't fails for debugging !
-    logNodeConfig.log(nodeConfigsToWrite.values.toSeq) match {
-      case eb:EmptyBox =>
-        val e = eb ?~! "Error when trying to write node configurations for debugging"
-        logger.error(e)
-        e.rootExceptionCause.foreach { ex =>
-          logger.error("Root exception cause was:", ex)
-        }
-      case _ => //nothing to do
+    val logNodeConfigurations = ZIO.when(logNodeConfig.isDebugEnabled) {
```

Comment:
this one must also help a lot - we were always duplicating all the nodeconfig written